src/libraries/System.Reflection.Context/Directory.Build.props

```diff
@@ -1,6 +1,9 @@
 ﻿<Project>
   <Import Project="..\Directory.Build.props" />
   <PropertyGroup>
+    <!-- this assembly is inbox in desktop, do not version it unless you
+         plan on shipping a new desktop version out of band. Instead add API
+         to a different assembly. -->
     <AssemblyVersion>4.0.3.0</AssemblyVersion>
```

Comment:
@ericstj do you know why this library is versioned higher than its .NET Framework runtime counterpart?

 > "C:\Windows\Microsoft.NET\Framework\v4.0.30319\System.Reflection.context.dll"

`[assembly: AssemblyVersion("4.0.0.0")]`

Reply:
Maybe because it was OOBed?  

.NET's unification table will shave off the last two parts of the version so this reference will still unify to the 4.0.0.0 version at runtime.  Not sure what it does at compile time.  It might be handled by RAR's AutoUnify flag.

Oddly it was even 4.0.4.0 in 3.1 https://github.com/dotnet/corefx/blob/b58ef0a19eca6269799664d6288be14b7f0cef3e/src/System.Reflection.Context/Directory.Build.props#L4

Reply:
As I mentioned in the other comment this same property was overridden in the ref-project to be set as `4.0.0.0`.  It looks like @ViktorHofer did this in https://github.com/dotnet/corefx/pull/29347.

At that time it was OK to do that, since we still shipped reference assemblies.  In 6.0 when we stopped shipping reference assemblies this higher version was exposed.

For the most part it's fine and doesn't hurt anything at runtime.  .NETFramework will still load this type just fine due to unification as I mentioned.

There is a warning from RAR on .NETFramework due to this mismatch, but it's just a warning.

We'd actually make things a bit worse if we went backwards on this version (at least in any runtime assembly). 

One thing we could consider -- if we merged this change in 8.0 to only pin the version for the `netstandard2.0` assembly - we could rollback the version to `4.0.0.0`.  That would fix the warning in the netframework/netstandard2.0 case, at the cost of introducing the same warning for people in the case of compiling against the old package in one library and using it in project referencing the new one.  Hard to say if it's worth the effort.  Regardless I'll open an issue for that: https://github.com/dotnet/runtime/issues/84320